Microsoft.Toolkit.Uwp.SampleApp/Shell.xaml.cs

```diff
@@ -514,10 +520,105 @@ private void HamburgerMenu_OnOptionsItemClick(object sender, ItemClickEventArgs
         {
-            e.Image = new BitmapImage(new Uri("ms-appx:///Assets/pixel.png"));
-            e.Handled = true;
+            async Task<Stream> CopyStream(HttpContent source)
+            {
+                var stream = new MemoryStream();
+                await source.CopyToAsync(stream);
+                stream.Seek(0, SeekOrigin.Begin);
+                return stream;
+            }
+
+            var deferral = e.GetDeferral();
+            BitmapImage image = null;
+
+            var absolute = Uri.TryCreate(e.Url, UriKind.Absolute, out Uri Url);
+            if (!absolute)
+            {
+                Url = new Uri(documentationPath + e.Url);
+            }
+
+            if (Url.Scheme == "ms-appx")
+            {
+                image = new BitmapImage(Url);
+            }
+            else
+            {
+                IRandomAccessStream imageStream = null;
+                var localpath = $"{Url.Host}/{Url.LocalPath}";
+
+                // Cache only in Release
+#if !DEBUG
+                try
+                {
+                    imageStream = await StreamHelper.GetLocalCacheFileStreamAsync(localpath, Windows.Storage.FileAccessMode.Read);
+                }
+                catch
+                {
+                }
+#endif
+
+                if (imageStream == null)
+                {
+                    try
+                    {
+                        using (var client = new HttpClient())
+                        {
+                            using (var response = await client.GetAsync(Url))
+                            {
+                                if (response.IsSuccessStatusCode)
+                                {
+                                    var imageCopy = await CopyStream(response.Content);
+                                    imageStream = imageCopy.AsRandomAccessStream();
+
+                                    // Cache only in Release
+#if !DEBUG
+                                    // Saves to the Cache.
+                                    SaveImageToCache(localpath, await CopyStream(response.Content));
+#endif
+                                }
+                            }
+                        }
+                    }
+                    catch
+                    {
+                    }
+                }
+
+                if (imageStream != null)
+                {
+                    image = new BitmapImage();
+                    await image.SetSourceAsync(imageStream);
+                }
+            }
+
+            // Handle only if no exceptions occur.
+            if (image != null)
+            {
+                e.Image = image;
+                e.Handled = true;
+            }
+
+            deferral.Complete();
+        }
+
+        private async void SaveImageToCache(string localpath, Stream imageStream)
+        {
+            // Close stream when finished.
+            using (imageStream)
```

Comment:
Move the using outside of the method. Methods usually will not dispose of objects passed to them

Reply:
It is inside the method, because we want to save the file asynchronously without interrupting the UI thread, so it takes a second copy of the image stream and saves it in this void method, this way it can display the image before the file is saved to improve performance.